pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -182,6 +183,24 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		if cosmosdb.IsErrorStatusCode(err, http.StatusPreconditionFailed) {
 			return nil, f.validateOpenShiftUniqueKey(ctx, doc)
 		}
+
+		now := time.Now().UTC()
```

Comment:
In other places we use cosmos triggers to set time values to provide an extra level of protection against time drift on our VMs.  I'd recommend doing the same for consistency.